src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    if (this.length === 0) {
+      return str;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      const expression = (this[i] === null || this[i] === undefined)
+        ? ''
+        : this[i];
+
+      if (i !== this.length - 1) {
+        str += expression + String(separator);
```

Comment:
when you add string (expression) to anything on the separator (for example, a number) it will convert it to string automatically